logger/src/main/java/com/orhanobut/logger/FileLogger.java

```diff
@@ -0,0 +1,362 @@
+
+            }
+            return;
+        }
+
+        if (message.contains(NEW_LINE)) {
+            message = message.replace(NEW_LINE, NEW_LINE_REPLACEMENT);
+        }
+
+        message = message.replace(HORZ_LINE, "").trim();
+
+        // messages
+        stringBuilder.append(separator);
+        stringBuilder.append(message);
+
+    }
+    //endregion
+
+    //region helper to generate file names
+    private static File getLogFile(String folderName, String fileName, int maxLines) {
+
+        File folder = new File(folderName);
+        if (!folder.exists()) {
+            folder.mkdirs();
+        }
+
+        int newFileCount = 0;
+        File newFile;
+        File existingFile = null;
+
+        newFile = new File(folder, String.format("%s_%s.csv", fileName, newFileCount));
+        while (newFile.exists()) {
+            existingFile = newFile;
+            newFileCount++;
+            newFile = new File(folder, String.format("%s_%s.csv", fileName, newFileCount));
+        }
+
+        if (existingFile != null) {
+            try {
+                int lineCount = 0;
+                BufferedReader br = new BufferedReader(new FileReader(existingFile));
+                while (br.readLine() != null) {
+                    lineCount++;
+                }
+
+                if (lineCount >= maxLines) {
+                    return newFile;
+                } else {
+                    return existingFile;
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        return newFile;
+    }
+    //endregion
+
+    //region handler to write to disk on background thread
+    private static class WriteHandler extends Handler {
+
+        private final String folder;
+        private final int maxLines;
+
+        private WriteHandler(Looper looper, String folder, int maxLines) {
+            super(looper);
+            this.folder = folder;
+            this.maxLines = maxLines;
+        }
+
+        @Override
+        public void handleMessage(Message msg) {
+            String text = (String) msg.obj;
+
+            FileWriter fileWriter = null;
+            File logFile = getLogFile(folder, "logs", maxLines);
+
+            try {
+                fileWriter = new FileWriter(logFile, true);
+                fileWriter.append(text);
+                fileWriter.flush();
+                fileWriter.close();
+            } catch (IOException e) {
+                if (fileWriter != null) {
+                    try {
+                        fileWriter.flush();
+                        fileWriter.close();
+                    } catch (IOException e1) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        }
+    }
+    //endregion
+
+    //region constructor/builder (mostly auto-generated with lombok and copied over here)
+    public static Builder builder() {
```

Comment:
`Looper` absolutely sure, as it's useful for a larger application to just re-use whichever thread it's being used for I/O operations.
`log level` as discussed previously it's also a very important piece of file logging
`folder` it's also very important, as it does matters for developers where those logs are going to. E.g. my use case I was marking folder with build number so I knew which log were from exact which version of the code.
The others I can see are not so important and we could use some default implementations.

Maybe a way around it could be have an empty constructor which would apply all the defaults, but still leave space for a builder.